conf/cassandra.yaml

```diff
@@ -589,6 +589,10 @@ commitlog_segment_size: 32MiB
 #     parameters:
 #         -
 
+# Configure commitlog disk access mode preferred order:direct_io, mmap, standard.
+# Default access mode is mmap. Direct I/O uses minimum page size to flush to disk.
+commitlog_disk_access_mode : mmap
```

Comment:
Given what @blambov mentioned and the test results - is there any reason to not immediately deprecate this setting in favour of hardcoding it to direct in the next release?

Reply:
> Given what @blambov mentioned and the test results - is there any reason to not immediately deprecate this setting in favour of hardcoding it to direct in the next release?

As per the earlier suggestion, I have emailed to dev-community to understand how to proceed. Link for the thread is [https://lists.apache.org/thread/t6v0p10737p0joob2vcsdt0r3g8zt94q](url). Some replies showed interest in Mmap as default to prevent any surprises. I want to share another point here. Direct-IO didn't work with Ramdisk while doing performance study. So, I feel it is better to have configuration option and default can be changed based on community feedback.

Reply:
Received another feedback yesterday regarding changing default to Direct-IO email and that makes it to 7 responses.

Summary:
1. mmap     :3/7
2. Direct-IO: 4/7

@blambov and @jacek-lewandowski your input might be helpful or please suggest what should be the default?

Reply:
My personal preference would be not only to change the default, but to scrap memory-mapped logs altogether. However,  for things like these we can't change the default unless everyone agrees or there's no expected downside. In this case a downside is the fact that a memory-mapped log might effectively persist data more often in the default periodic mode.

`mmap` should thus remain default, with a switch to `direct` as one of the items in the optimized configuration envisioned by CASSANDRA-18753.

Reply:
I agree with @blambov